api/server/server.go

```diff
@@ -800,11 +807,14 @@ func (s *Server) postImagesCreate(eng *engine.Engine, version version.Version, w
 		imageImportConfig.ContainerConfig = newConfig
 
 		if err := s.daemon.Repositories().Import(src, repo, tag, imageImportConfig); err != nil {
-			return err
+			if !output.Flushed() {
```

Comment:
ping @vieux as original author of such trick. Do we need it for import and push?